src/network/tarcap/node_stats.cpp

```diff
@@ -0,0 +1,216 @@
+#include "node_stats.hpp"
+
+#include "consensus/pbft_chain.hpp"
+#include "consensus/pbft_manager.hpp"
+#include "consensus/vote.hpp"
+#include "dag/dag.hpp"
+#include "dag/dag_block_manager.hpp"
+#include "libp2p/Common.h"
+#include "network/tarcap/packets_handler/peers_state.hpp"
+#include "network/tarcap/packets_handler/syncing_state.hpp"
+#include "transaction_manager/transaction_manager.hpp"
+
+namespace taraxa::network::tarcap {
+
+NodeStats::NodeStats(std::shared_ptr<PeersState> peers_state, std::shared_ptr<SyncingState> syncing_state,
+                     std::shared_ptr<PbftChain> pbft_chain, std::shared_ptr<PbftManager> pbft_mgr,
+                     std::shared_ptr<DagManager> dag_mgr, std::shared_ptr<DagBlockManager> dag_blk_mgr,
+                     std::shared_ptr<VoteManager> vote_mgr, std::shared_ptr<TransactionManager> trx_mgr,
+                     uint64_t stats_log_interval, const addr_t &node_addr)
+    : peers_state_(std::move(peers_state)),
+      syncing_state_(std::move(syncing_state)),
+      pbft_chain_(std::move(pbft_chain)),
+      pbft_mgr_(std::move(pbft_mgr)),
+      dag_mgr_(std::move(dag_mgr)),
+      dag_blk_mgr_(std::move(dag_blk_mgr)),
+      vote_mgr_(std::move(vote_mgr)),
+      trx_mgr_(std::move(trx_mgr)),
+      stats_log_interval_(stats_log_interval) {
+  LOG_OBJECTS_CREATE("SUMMARY");
+}
+
+uint64_t NodeStats::getNodeStatsLogInterval() const { return stats_log_interval_; }
+
+uint64_t NodeStats::syncTimeSeconds() const { return stats_log_interval_ * syncing_interval_count_ / 1000; }
+
+void NodeStats::logNodeStats() {
+  bool is_syncing = syncing_state_->is_syncing();
+
+  dev::p2p::NodeID max_pbft_round_nodeID;
```

Comment:
naming -> max_pbft_round_node_id

Reply:
everything is copy paste but yeah I agree that I can refactor bad naming here...

Reply:
fixed